connectors/stash/src/main/java/com/vmware/connectors/stash/StashController.java

```diff
@@ -0,0 +1,444 @@
+/*
+ * Copyright © 2018 VMware, Inc. All Rights Reserved.
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+package com.vmware.connectors.stash;
+
+import com.google.common.collect.ImmutableMap;
+import com.vmware.connectors.common.json.JsonDocument;
+import com.vmware.connectors.common.payloads.request.CardRequest;
+import com.vmware.connectors.common.payloads.response.*;
+import com.vmware.connectors.common.utils.Async;
+import com.vmware.connectors.common.utils.CardTextAccessor;
+import com.vmware.connectors.common.web.ObservableUtil;
+import com.vmware.connectors.stash.utils.StashAction;
+import com.vmware.connectors.stash.utils.StashComment;
+import com.vmware.connectors.stash.utils.StashConstants;
+import com.vmware.connectors.stash.utils.StashPullRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.*;
+import org.springframework.util.Assert;
+import org.springframework.util.CollectionUtils;
+import org.springframework.util.concurrent.ListenableFuture;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.client.AsyncRestOperations;
+import rx.Observable;
+import rx.Single;
+
+import javax.annotation.Resource;
+import javax.validation.Valid;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import static com.vmware.connectors.stash.utils.StashConstants.ATLASSIAN_TOKEN;
+import static com.vmware.connectors.stash.utils.StashConstants.COMMENT_PARAM_KEY;
+import static org.springframework.http.HttpHeaders.AUTHORIZATION;
+
+@RestController
+public class StashController {
```

Comment:
Since Bitbucket is using Basic auth, you'll want to implement a testAuth method for it like Jira:

* Example implementation of test-auth method: https://github.com/vmware/connectors-workspace-one/blob/master/connectors/jira/src/main/java/com/vmware/connectors/jira/JiraController.java#L129
* Example configuration in yml: https://github.com/vmware/connectors-workspace-one/blob/master/connectors/jira/src/main/resources/application.yml#L8

Reply:
This allows the client to call into the server and verify the credentials that the user typed in were correct immediately rather than at a later point when a card is fetched.

Reply:
I have added ```test-auth``` api to BitbucketController to test basic auth..